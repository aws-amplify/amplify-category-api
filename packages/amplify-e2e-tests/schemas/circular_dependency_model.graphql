type Member @model @auth(rules: [{ allow: owner }])
{
  owner: String! @primaryKey
  secondaryIndex: String @index(name: "bySecondaryIndex", queryField: "memberBySecondaryIndex")
  washCarTasks: [WashCarTask] @hasMany(indexName: "byOwner", fields: ["owner"])
}

type WashCarTask @model @auth(rules: [{ allow: owner }]) {
  id: ID! @primaryKey
  owner: String! @index(name: "byOwner", queryField: "washCarTaskByOwner")
}