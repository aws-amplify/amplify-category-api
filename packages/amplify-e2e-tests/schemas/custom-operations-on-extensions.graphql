type Query {
  myFunctionQueryBase: String @function(name: "myFunctionQueryBase-${env}") @auth(rules: [{ allow: custom }])
  myHttpQueryBase: String @http(url: "<REST_URL>") @auth(rules: [{ allow: custom }])
}

extend type Query {
  myFunctionQueryExtended: String @function(name: "myFunctionQueryExtended-${env}") @auth(rules: [{ allow: custom }])
  myHttpQueryExtended: String @http(url: "<REST_URL>") @auth(rules: [{ allow: custom }])
}

type Mutation {
  myFunctionMutationBase: String @function(name: "myFunctionQueryBase-${env}") @auth(rules: [{ allow: custom }])
  myHttpMutationBase: String @http(url: "<REST_URL>") @auth(rules: [{ allow: custom }])
}

extend type Mutation {
  myFunctionMutationExtended: String @function(name: "myFunctionQueryExtended-${env}") @auth(rules: [{ allow: custom }])
  myHttpMutationExtended: String @http(url: "<REST_URL>") @auth(rules: [{ allow: custom }])
}
