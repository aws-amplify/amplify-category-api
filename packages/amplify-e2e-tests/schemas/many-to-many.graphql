type Post @model @auth(rules: [{ allow: public }]) {
  id: ID!
  title: String!
  content: String
  tags: [Tag] @manyToMany(relationName: "PostTags")
}

type Tag @model @auth(rules: [{ allow: public }]) {
  id: ID!
  label: String!
  posts: [Post] @manyToMany(relationName: "PostTags")
}
