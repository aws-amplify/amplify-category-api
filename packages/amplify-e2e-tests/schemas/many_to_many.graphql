type Post @model {
  id: ID!
  title: String!
  content: String
  tags: [Tag] @manyToMany(relationName: "PostTags")
}

type Tag @model {
  id: ID!
  label: String!
  posts: [Post] @manyToMany(relationName: "PostTags")
}
