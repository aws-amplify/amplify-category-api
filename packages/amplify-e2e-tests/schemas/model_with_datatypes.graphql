type Task @model @auth(rules: [{ allow: public }]) {
  id: ID! @primaryKey
  title: String!
  description: String
  priority: Int
  dueDate: AWSDateTime
}
