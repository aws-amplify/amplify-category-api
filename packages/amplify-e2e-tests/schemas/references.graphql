type Primary @model @auth(rules: [{ allow: public }]) {
  id: ID!
  relatedMany: [RelatedMany] @hasMany(references: "primaryId")
  relatedOne: RelatedOne @hasOne(references: "primaryId")
}

type RelatedMany @model @auth(rules: [{ allow: public }]) {
  id: ID!
  primaryId: String
  primary: Primary @belongsTo(references: ["primaryId"])
}

type RelatedOne @model @auth(rules: [{ allow: public }]) {
  id: ID!
  primaryId: String
  primary: Primary @belongsTo(references: ["primaryId"])
}
