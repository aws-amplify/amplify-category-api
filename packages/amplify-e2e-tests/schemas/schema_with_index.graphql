input AMPLIFY {
  globalAuthRule: AuthRule = { allow: public }
}

type Song @model {
  id: ID!
  name: String!
  genre: String! @index(name: "byGenre", queryField: "songInfoByGenre")
}

type SongWithSortKey @model {
  id: ID!
  name: String! @index(name: "byNameAndGenre", sortKeyFields: ["genre"])
  genre: String!
}
