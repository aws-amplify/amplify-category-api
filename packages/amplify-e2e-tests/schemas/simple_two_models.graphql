type Todo @model {
  id: ID!
  content: String
}
type Author @model {
  id: ID!
  name: String
}
