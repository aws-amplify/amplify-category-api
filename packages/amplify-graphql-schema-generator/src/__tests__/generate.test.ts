import { ImportedRDSType } from '@aws-amplify/graphql-transformer-core';
import { graphqlSchemaFromRDSSchema } from '..';
import { schemas } from './__utils__/schemas';

describe('generate', () => {
  describe('graphqlSchemaFromRDSSchema', () => {
    const cases = Object.entries(schemas)
      .map(([engineType, schemasForEngine]) => Object.entries(schemasForEngine).map((schema) => [engineType, ...schema]))
      .reduce((accumulator, value) => accumulator.concat(value), []);
    test.each(cases)('creates graphql schema from %p %p schema', (engineType, _, schema) => {
      expect(graphqlSchemaFromRDSSchema(schema, engineType as ImportedRDSType)).toMatchSnapshot();
    });
  });
});
