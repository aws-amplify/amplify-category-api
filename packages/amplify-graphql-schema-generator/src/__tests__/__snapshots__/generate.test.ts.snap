// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate graphqlSchemaFromRDSSchema creates graphql schema from "mysql" "news" schema 1`] = `
"input AMPLIFY {
 engine: String = \\"mysql\\"  
}


type Article @refersTo(name: \\"articles\\") @model {
  pubId: Int! @refersTo(name: \\"pub_id\\") @default(value: \\"NULL\\") @primaryKey(sortKeyFields: [\\"pubType\\"])
  pubType: String! @refersTo(name: \\"pub_type\\") @default(value: \\"A\\")
  placeholder: String @default(value: \\"NULL\\") @index(name: \\"NULL\\")
}

type Comment @refersTo(name: \\"comments\\") @model {
  pubId: Int! @refersTo(name: \\"pub_id\\") @default(value: \\"NULL\\") @primaryKey(sortKeyFields: [\\"commentTimestamp\\", \\"commenterEmail\\"])
  commentTimestamp: AWSDateTime! @refersTo(name: \\"comment_timestamp\\") @default(value: \\"CURRENT_TIMESTAMP\\")
  commenterEmail: String! @refersTo(name: \\"commenter_email\\") @default(value: \\"NULL\\")
  commentText: String @refersTo(name: \\"comment_text\\") @default(value: \\"NULL\\") @index(name: \\"NULL\\")
}

type Publication @refersTo(name: \\"publications\\") @model {
  pubId: Int! @refersTo(name: \\"pub_id\\") @default(value: \\"NULL\\") @primaryKey @index(name: \\"publications_superkey\\", sortKeyFields: [\\"pubType\\"])
  pubType: String @refersTo(name: \\"pub_type\\") @default(value: \\"NULL\\")
  pubUrl: String @refersTo(name: \\"pub_url\\") @default(value: \\"NULL\\") @index(name: \\"pub_url\\")
}

type Story @refersTo(name: \\"stories\\") @model {
  pubId: Int! @refersTo(name: \\"pub_id\\") @default(value: \\"NULL\\") @primaryKey(sortKeyFields: [\\"pubType\\"])
  pubType: String! @refersTo(name: \\"pub_type\\") @default(value: \\"S\\")
  placeholder: String @default(value: \\"NULL\\") @index(name: \\"NULL\\")
}
"
`;

exports[`generate graphqlSchemaFromRDSSchema creates graphql schema from "mysql" "todo" schema 1`] = `
"input AMPLIFY {
 engine: String = \\"mysql\\"  
}


type Todo @model {
  id: Int! @default(value: \\"NULL\\") @primaryKey
  title: String @default(value: \\"NULL\\")
  startDate: AWSDate @refersTo(name: \\"start_date\\") @default(value: \\"NULL\\")
  dueDate: AWSDate @refersTo(name: \\"due_date\\") @default(value: \\"NULL\\") @index(name: \\"NULL\\", sortKeyFields: [\\"startDate\\", \\"title\\"])
}
"
`;

exports[`generate graphqlSchemaFromRDSSchema creates graphql schema from "postgres" "news" schema 1`] = `
"input AMPLIFY {
 engine: String = \\"postgres\\"  
}


type Article @refersTo(name: \\"articles\\") @model {
  pubId: Int! @refersTo(name: \\"pub_id\\") @default(value: \\"NULL\\") @primaryKey(sortKeyFields: [\\"pubType\\"])
  pubType: String! @refersTo(name: \\"pub_type\\") @default(value: \\"'A'::bpchar\\")
  placeholder: String @default(value: \\"NULL\\")
}

type Comment @refersTo(name: \\"comments\\") @model {
  pubId: Int! @refersTo(name: \\"pub_id\\") @default(value: \\"NULL\\") @primaryKey(sortKeyFields: [\\"commentTimestamp\\", \\"commenterEmail\\"])
  commentTimestamp: AWSDateTime! @refersTo(name: \\"comment_timestamp\\")
  commenterEmail: String! @refersTo(name: \\"commenter_email\\") @default(value: \\"NULL\\")
  commentText: String @refersTo(name: \\"comment_text\\") @default(value: \\"NULL\\")
}

type Publication @refersTo(name: \\"publications\\") @model {
  pubId: Int! @refersTo(name: \\"pub_id\\") @default(value: \\"NULL\\") @primaryKey @index(name: \\"publications_superkey\\", sortKeyFields: [\\"pubType\\"])
  pubType: String @refersTo(name: \\"pub_type\\") @default(value: \\"NULL\\")
  pubUrl: String @refersTo(name: \\"pub_url\\") @default(value: \\"NULL\\") @index(name: \\"publications_pub_url_key\\")
}

type Story @refersTo(name: \\"stories\\") @model {
  pubId: Int! @refersTo(name: \\"pub_id\\") @default(value: \\"NULL\\") @primaryKey(sortKeyFields: [\\"pubType\\"])
  pubType: String! @refersTo(name: \\"pub_type\\") @default(value: \\"'S'::bpchar\\")
  placeholder: String @default(value: \\"NULL\\")
}
"
`;

exports[`generate graphqlSchemaFromRDSSchema creates graphql schema from "postgres" "todo" schema 1`] = `
"input AMPLIFY {
 engine: String = \\"postgres\\"  
}


type Todo @refersTo(name: \\"todo\\") @model {
  id: Int! @default(value: \\"NULL\\") @primaryKey
  title: String @default(value: \\"NULL\\")
  startDate: AWSDate @refersTo(name: \\"start_date\\") @default(value: \\"NULL\\")
  dueDate: AWSDate @refersTo(name: \\"due_date\\") @default(value: \\"NULL\\")
}
"
`;
